llvm/lib/CodeGen/RegAllocBase.h

```diff
@@ -98,6 +101,9 @@ class RegAllocBase {
   /// enqueue - Add VirtReg to the priority queue of unassigned registers.
   virtual void enqueueImpl(const LiveInterval *LI) = 0;
 
+  /// Hook for allocators to initialize state for already-allocated registers.
+  virtual void noteAllocatedReg(const LiveInterval *LI) {}
```

Comment:
This is because vregs with an existing allocation in the `VirtRegMap` are ignored by Greedy and are not given a `Stage`. We want them to have one so that they can be part of the standard procedures, and e.g. be split/evicted.